packages/swap/src/event-handlers/swapDepositAddressReady.ts

```diff
@@ -0,0 +1,57 @@
+import { z } from 'zod';
+import { u64, chainflipAssetEnum } from '@/shared/parsers';
+import { encodedAddress } from './common';
+import { EventHandlerArgs } from './index';
+
+const swapDepositAddressReadyArgs = z.object({
+  depositAddress: encodedAddress,
+  destinationAddress: encodedAddress,
+  sourceAsset: chainflipAssetEnum,
+  destinationAsset: chainflipAssetEnum,
+  channelId: u64,
+});
+
+export type SwapDepositAddressReadyArgs = z.input<
+  typeof swapDepositAddressReadyArgs
+>;
+
+export const swapDepositAddressReady = async ({
+  prisma,
+  event,
+  block,
+}: EventHandlerArgs): Promise<void> => {
+  const issuedBlock = block.height;
+
+  const {
+    depositAddress,
+    destinationAddress,
+    sourceAsset,
+    destinationAsset,
+    channelId,
+    ...rest
+  } = swapDepositAddressReadyArgs.parse(event.args);
+
+  await prisma.swapDepositChannel.upsert({
```

Comment:
We should probably change the other insertion into an upsert too

Reply:
Which one? Can you point me to it please?

Reply:
Which one? Can you point me to it please?

Reply:
Done